packages/scheduler/src/Scheduler.js

```diff
@@ -92,57 +90,56 @@ function flushTask(task) {
 
   // Now it's safe to execute the task.
   var callback = task.callback;
-  var expirationTime = task.expirationTime;
-  var priorityLevel = task.priorityLevel;
   var previousPriorityLevel = currentPriorityLevel;
-  var previousExpirationTime = currentExpirationTime;
-  currentPriorityLevel = priorityLevel;
-  currentExpirationTime = expirationTime;
+  var previousTask = currentTask;
+  currentPriorityLevel = task.priorityLevel;
+  currentTask = task;
   var continuationCallback;
   try {
-    const didUserCallbackTimeout =
+    var didUserCallbackTimeout =
       currentHostCallbackDidTimeout ||
       // Immediate priority callbacks are always called as if they timed out
-      priorityLevel === ImmediatePriority;
+      currentPriorityLevel === ImmediatePriority;
     continuationCallback = callback(didUserCallbackTimeout);
   } catch (error) {
     throw error;
   } finally {
     currentPriorityLevel = previousPriorityLevel;
-    currentExpirationTime = previousExpirationTime;
+    currentTask = previousTask;
   }
 
   // A callback may return a continuation. The continuation should be scheduled
   // with the same priority and expiration as the just-finished callback.
   if (typeof continuationCallback === 'function') {
     var continuationTask = {
       callback: continuationCallback,
-      priorityLevel,
-      expirationTime,
+      priorityLevel: task.priorityLevel,
+      startTime: task.startTime,
+      timeout: task.timeout,
       next: null,
       previous: null,
     };
 
-    // Insert the new callback into the list, sorted by its expiration. This is
+    // Insert the new callback into the list, sorted by its timeout. This is
     // almost the same as the code in `scheduleCallback`, except the callback
-    // is inserted into the list *before* callbacks of equal expiration instead
+    // is inserted into the list *before* callbacks of equal timeout instead
     // of after.
+    var absoluteTimeout = continuationTask.startTime + continuationTask.timeout;
     if (firstTask === null) {
       // This is the first callback in the list.
       firstTask = continuationTask.next = continuationTask.previous = continuationTask;
     } else {
       var nextAfterContinuation = null;
-      var task = firstTask;
+      var t = firstTask;
       do {
-        if (task.expirationTime >= expirationTime) {
-          // This task expires at or after the continuation. We will insert the
-          // continuation *before* this task.
-          nextAfterContinuation = task;
+        if (absoluteTimeout <= t.startTime + t.timeout) {
```

Comment:
This forces you to do a computation each time you want to check it instead of only once. Is there a reason you don't just do startTime and expirationTime? In case the task doesn't gets added to the queue at exactly the startTime? In that case it seems like it should be increasing in priority since it's running behind.

Reply:
Yeah that makes sense. I think originally I thought I would the timeout after creation.